src/vec3.rs

```diff
@@ -193,6 +193,28 @@ macro_rules! impl_vec3_float_methods {
         pub fn angle_between(self, other: Self) -> $t {
             self.0.angle_between(other.0)
         }
+
+        /// Returns somes vector that is orthogonal to the given one.
+        ///
+        /// The output vector is not necessarily unit-length.
+        /// For that use [`Self::any_orthonormal`] instead.
+        ///
+        /// The input vector must be finite and non-zero.
+        #[inline]
+        pub fn any_orthogonal(&self) -> Self {
```

Comment:
I haven't come across these before, I'm curious what they are used for?

Reply:
For the `from_rotation_arc` I use it to pick an axis of rotation in the case of a 180° singularity. That is, to rotate from `+v` to `-v` you can pick any axis that is orthogonal to `v`. 

There are many other cases where it is useful to be able to create an orthonormal basis. For instance you may have a surface normal but want to be able to create an ad-hoc tangent space, so you need a tangent. There are an infnite number of tangents to pick from, so: `let tangent = normal.any_orthonormal();`.

My implementation here isn't the most efficient though. I should probably try out this method instead: https://graphics.pixar.com/library/OrthonormalB/paper.pdf (especially for the `any_orthonormal()` method).

If it's worth doing, it's worth doing right. I'll make some additional commits improving this (and add a benchmark)

Reply:
Cool, thanks for the explanation 👍